app/src/main/java/com/tomtom/sdk/examples/maps/ConfigurableMapActivity.kt

```diff
@@ -167,9 +191,136 @@ class ConfigurableMapActivity : AppCompatActivity() {
+        bottomOptionsDialog.setContentView(mapOptionsDialogBinding.root)
+        bottomOptionsDialog.show()
+
+        mapOptionsDialogBinding.optionsTitleTextView.setOnClickListener {
+            bottomOptionsDialog.dismiss()
+        }
+
+        bottomOptionsDialog.setOnDismissListener {
+            if(isMapChanged) {
+                upgradeMap()
+            }
+            if(areMapPreferencesChanged) {
+                upgradeMapPreferences()
+            }
+            isMapChanged = false
+            areMapPreferencesChanged = false
+        }
+
+        initBaseMapStyleList()
+        initMapPreferencesList()
+        initCurrentMapPreferences()
+
+        val baseMapStyleRecyclerList: RecyclerView = mapOptionsDialogBinding.baseMapStylesRecyclerList
+        val mapPreferenceRecyclerList: RecyclerView = mapOptionsDialogBinding.mapPreferencesRecyclerList
+
+        setUpRecyclerListForBaseMapStyle(baseMapStyleRecyclerList)
+        setUpRecyclerListForMapPreferences(mapPreferenceRecyclerList)
+    }
+
+    private val listener: OnRecyclerViewItemClickListener = object : OnRecyclerViewItemClickListener {
+        override fun onBaseMapStyleItemClick(baseMapStyleItem: BaseMapStyleItem) {
+            if(currentBaseMapStyle != baseMapStyleItem.styleMode || currentMapStyleDescriptor != baseMapStyleItem.styleDescriptor) {
+                mapBaseStyleToUpgrade = baseMapStyleItem.styleMode
+                mapStyleDescriptorToUpgrade = baseMapStyleItem.styleDescriptor
+                isMapChanged = true
+            } else {
+                isMapChanged = false
+            }
+        }
+
+        override fun onMapPreferenceItemClick(title: String, mapPreferenceItemMethod: MapPreference) {
+            areMapPreferencesChanged = if(!currentMapPreferences.containsValue(mapPreferenceItemMethod)) {
+                mapPreferencesToUpgrade[title] = mapPreferenceItemMethod
+                true
+            } else {
+                if(mapPreferencesToUpgrade.containsKey(title)) {
+                    mapPreferencesToUpgrade.remove(title)
+                }
+                false
+            }
+        }
+    }
+
+    private fun upgradeMap() {
+        val onStyleLoadedCallback = object: StyleLoadingCallback {
+            override fun onSuccess() {
+                currentMapStyleDescriptor = mapStyleDescriptorToUpgrade
+            }
+
+            override fun onFailure(failure: LoadingStyleFailure) {
+                //Log.d() a message
+            }
+        }
+        tomTomMap.setStyleMode(mapBaseStyleToUpgrade)
+        tomTomMap.loadStyle(mapStyleDescriptorToUpgrade, onStyleLoadedCallback)
+        currentBaseMapStyle = mapBaseStyleToUpgrade
+    }
+
+    private fun upgradeMapPreferences() {
+        mapPreferencesToUpgrade.forEach { (key, value) ->
+            val method = tomTomMap.javaClass.getMethod(value.methodName)
+            method.invoke(tomTomMap)
+            currentMapPreferences[key] = value
```

Comment:
This is an example app for the customers to understand how to use the SDK, so calling API methods using reflection is too complex for the example